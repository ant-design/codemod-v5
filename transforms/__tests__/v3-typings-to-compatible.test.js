jest.mock('../v3-typings-to-compatible', () => {
  return Object.assign(require.requireActual('../v3-typings-to-compatible'), {
    parser: 'babylon',
  });
});

const defineSnapshotTest = require('jscodeshift/dist/testUtils')
  .defineSnapshotTest;

const testUnit = 'v3-typings-to-compatible';

const transform = require(`../${testUnit}`);

describe(testUnit, () => {
  defineSnapshotTest(
    transform,
    {},
    `import { FormComponentProps, WrappedFormUtils } from 'antd/es/form';`,
    'basic es import',
  );

  defineSnapshotTest(
    transform,
    {},
    `import { FormComponentProps, WrappedFormUtils } from 'antd/lib/form';`,
    'basic lib import',
  );

  defineSnapshotTest(
    transform,
    {},
    `import { FormComponentProps as AAFormProps, WrappedFormUtils as AAWrappedFormUtils } from 'antd/es/form';`,
    'alias import from es',
  );

  defineSnapshotTest(
    transform,
    {},
    `import { FormComponentProps as AAFormProps, WrappedFormUtils as AAWrappedFormUtils } from 'antd/lib/form';`,
    'alias import from lib',
  );

  defineSnapshotTest(
    transform,
    {},
    `import { FormComponentProps, FormItemProps, WrappedFormUtils } from 'antd/lib/form';`,
    'basic import: multi',
  );

  defineSnapshotTest(
    transform,
    {},
    `import { FormComponentProps as AAFormProps, FormItemProps as AAFormItemProps } from 'antd/lib/form';`,
    'alias import: multi',
  );
});
