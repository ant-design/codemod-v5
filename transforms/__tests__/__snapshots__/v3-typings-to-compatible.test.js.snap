// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v3-typings-to-compatible alias import from es 1`] = `"import { FormComponentProps as AAFormProps, WrappedFormUtils as AAWrappedFormUtils } from '@ant-design/compatible/es/form';"`;

exports[`v3-typings-to-compatible alias import from lib 1`] = `"import { FormComponentProps as AAFormProps, WrappedFormUtils as AAWrappedFormUtils } from '@ant-design/compatible/lib/form';"`;

exports[`v3-typings-to-compatible alias import: multi 1`] = `
"import { FormComponentProps as AAFormProps } from '@ant-design/compatible/lib/form';
import { FormItemProps as AAFormItemProps } from 'antd/lib/form';"
`;

exports[`v3-typings-to-compatible basic es import 1`] = `"import { FormComponentProps, WrappedFormUtils } from '@ant-design/compatible/es/form';"`;

exports[`v3-typings-to-compatible basic import: multi 1`] = `
"import { FormComponentProps, WrappedFormUtils } from '@ant-design/compatible/lib/form';
import { FormItemProps } from 'antd/lib/form';"
`;

exports[`v3-typings-to-compatible basic lib import 1`] = `"import { FormComponentProps, WrappedFormUtils } from '@ant-design/compatible/lib/form';"`;
